lib/file_transfer_agent/file_util.js

```diff
@@ -144,4 +146,22 @@ function getMatchingFilePaths(dir, fileName) {
   return glob.sync(pathWithWildcardDependsOnPlatform);
 }
 
+async function validateOnlyUserReadWritePermission(filePath) {
+  try {
+    await fsPromises.access(filePath, fs.constants.F_OK);
+  } catch (err) {
+    return;
+  }
+  const mode = (await fsPromises.stat(filePath)).mode;
+  const permission = mode & 0o600;
+
+  //This should be 600 permission, which means the file permission has not been changed by others.
+  if (permission.toString(8) === '600') {
+    Logger.getInstance().info('Validated that the user has read and write permission');
+  } else {
+    throw new Error('File permissions different than read/write for user.');
```

Comment:
what about execute?

Reply:
600 is only read nad write. User shouldn't have execute permission